autoload/nvimdev.vim

```diff
@@ -145,9 +147,6 @@ function! nvimdev#init(path) abort
   augroup nvimdev
     autocmd!
     autocmd BufRead,BufNewFile *.h set filetype=c
-    if get(g:, 'nvimdev_auto_lint', 1)
```

Comment:
TBH, I'm not that concerned about the interpretation of the variable name since it is documented in the README.  I do agree that it was a poor choice of name, though.

I'm fine with the changes, but could you update the README as well?

Let's also leave in the `autocmd` at least until neomake/neomake#1055 is merged.